Tailwind V4 Upgrade 💾

tnylea opened the pull request:
This PR includes the upgrade to Tailwind CSS v4.

You can find more details about the upgrade process here: https://www.shadcn-vue.com/docs/tailwind-v4.html. A few additional modifications have been made to the styles and sidebar to ensure everything looks good. I experienced a few layout issues that might have resulted from upgrading from ShadCN Vue 1 to v2 with the latest tailwind.

I reviewed the starter kit manually to ensure everything looks the same. Here's a quick screencast of these tests.

https://github.com/user-attachments/assets/e0a146c1-fe82-4eaa-8523-3adb9e507219

Everything seems to be working as it did before; however, this new version now includes the new Reka UI lib along with the latest version of Tailwind 👏

ibrunotome commented on vite.config.ts:
```diff
@@ -30,7 +31,7 @@ export default defineConfig({
     css: {
         postcss: {
-            plugins: [tailwindcss, autoprefixer],
+            plugins: [autoprefixer],
```
There is no need for autoprefixer in v4

tnylea replied:
thanks. I've updated that in this PR 👏

ibrunotome commented on resources/css/app.css:
```diff
@@ -1,84 +1,162 @@
+  }
+}
+
+:root {
+  --background: hsl(0 0% 100%);
+  --foreground: hsl(0 0% 3.9%);
+  --card: hsl(0 0% 100%);
+  --card-foreground: hsl(0 0% 3.9%);
+  --popover: hsl(0 0% 100%);
+  --popover-foreground: hsl(0 0% 3.9%);
+  --primary: hsl(0 0% 9%);
+  --primary-foreground: hsl(0 0% 98%);
+  --secondary: hsl(0 0% 92.1%);
+  --secondary-foreground: hsl(0 0% 9%);
+  --muted: hsl(0 0% 96.1%);
+  --muted-foreground: hsl(0 0% 45.1%);
+  --accent: hsl(0 0% 96.1%);
+  --accent-foreground: hsl(0 0% 9%);
+  --destructive: hsl(0 84.2% 60.2%);
+  --destructive-foreground: hsl(0 0% 98%);
+  --border: hsl(0 0% 92.8%);
+  --input: hsl(0 0% 89.8%);
+  --ring: hsl(0 0% 3.9%);
+  --chart-1: hsl(12 76% 61%);
+  --chart-2: hsl(173 58% 39%);
+  --chart-3: hsl(197 37% 24%);
+  --chart-4: hsl(43 74% 66%);
+  --chart-5: hsl(27 87% 67%);
+  --radius: 0.5rem;
+  --sidebar-background: hsl(0 0% 98%);
+  --sidebar-foreground: hsl(240 5.3% 26.1%);
+  --sidebar-primary: hsl(0 0% 10%);
+  --sidebar-primary-foreground: hsl(0 0% 98%);
+  --sidebar-accent: hsl(0 0% 94%);
+  --sidebar-accent-foreground: hsl(0 0% 30%);
+  --sidebar-border: hsl(0 0% 91%);
+  --sidebar-ring: hsl(217.2 91.2% 59.8%);
+}
```
I think that colors must be defined using oklch

tnylea replied:
In the upgrade documentation: https://www.shadcn-vue.com/docs/tailwind-v4, they are suggesting to wrap in `hsl` 👍

ibrunotome reviewed (commented):
Just a little comment about oklch colors and autoprefixer plugin.

Bosphoramus commented:
Hi @tnylea 

I made a similar PR where i also updated to shadcn/vue 2.0

I think it would be cool if you checked the changelog i provided there and maybe find a way to "merge" both PRs

amdad121 commented on vite.config.ts:
```diff
@@ -30,7 +31,7 @@ export default defineConfig({
     },
     css: {
         postcss: {
```
When you are using @tailwindcss/vite, you do not need to use postcss and autoprefixer.

tnylea replied:
Just updated that and removed from `vite.config.js` Thanks

Bosphoramus commented:
Yes, in `vite.config.js`

```js
css: {
    postcss: {
        plugins: [tailwindcss, autoprefixer],
    },
},
```

can be removed when using the Tailwind Vite plugin installation

Also, i think some shadcn components got updates in version 2 and are still not updated in this PR, it can be done running this in the console:

`npx shadcn-vue@latest add avatar breadcrumb button card checkbox collapsible dialog dropdown-menu input label navigation-menu separator sheet sidebar skeleton tooltip`

Those are all the components we are using for the Vue starter kit

Bosphoramus commented:
NPM dependencies `autoprefixer` and `tailwindcss-animate` can also be deleted, as i think `autoprefixer` is no longer required with the Tailwind Vite plugin installation

And shadcn/vue 2 now uses `tw-animate-css` instead as you can see [here](https://www.shadcn-vue.com/docs/installation/manual.html)

JustSteveKing commented:
This is going to be 🔥

tnylea commented:
Thanks @Bosphoramus for all the feedback.

It looks like many of the components have been updated since the first release of ShadCN Vue 2.0. I've updated them and re-checked that all the layouts look correct. I've also swapped out `tailwind-animate` with `tw-animate` and removed autoprefixer from the vite config.

Feel free to send over any additional feedback that you implemented in your version of the upgrade and I'll take a look into it.

Appreciate it!
